src/status_im/ui2/screens/common/contact_list/view.cljs

```diff
@@ -0,0 +1,50 @@
+(ns status-im.ui2.screens.common.contact-list.view
+  (:require [react-native.core :as rn]
+            [quo2.foundations.colors :as colors]
+            [status-im.ui2.screens.chat.components.contact-item.view :as contact-item]
+            [quo2.core :as quo2]
+            [utils.re-frame :as rf]
+            [reagent.core :as reagent]
+            [clojure.string :as str]))
+
+(def query (reagent/atom ""))
+
+(defn prepare-contacts [contacts]
+  (let [data (atom {})]
+    (doseq [i (range (count contacts))]
+      (let [first-char (get (:alias (nth contacts i)) 0)]
+        (when (or (empty? @query) (str/includes? (str/lower-case (:alias (nth contacts i))) (str/lower-case @query)))
+          (if-not (contains? @data first-char)
+            (swap! data #(assoc % first-char {:title first-char :data [(nth contacts i)]}))
+            (swap! data #(assoc-in % [first-char :data] (conj (:data (get @data first-char)) (nth contacts i))))))))
+    (swap! data #(sort @data))
+    (vals @data)))
+
+(defn contacts-section-header [{:keys [title]}]
+  [rn/view {:style {:border-top-width   1
+                    :border-top-color   colors/neutral-20
+                    :padding-vertical   8
+                    :padding-horizontal 20 :margin-top 8}}
+   [quo2/text {:size   :paragraph-2
+               :weight :medium
+               :style  {:color colors/neutral-50}} title]])
+
+(defn search-input []
+  [rn/text-input {:placeholder "Search..."
+                  :style       {:height             32
+                                :padding-horizontal 20
+                                :margin-vertical    12}
+                  :on-change   (fn [e] (reset! query (.-text ^js (.-nativeEvent ^js e))))}])
```

Comment:
Are you sure you need to add all the type checks like this? I believe it's enough to only add it to the function argument. So something like this seems better:

```clojure
(fn [^js e] (reset! query (.. e -nativeEvent -text)))
```

You could also use `oops.core/oget`.

Reply:
<img width="584" alt="image" src="https://user-images.githubusercontent.com/11790366/203537380-769e1e5f-9a2d-409e-8f79-ac28daa58d5d.png">
we have a bunch of different variants, it would be great to have it documented

Reply:
i like `(oget e "nativeEvent.text")`

Reply:
> we have a bunch of different variants, it would be great to have it documented

Great point @flexsurfer, I totally agree. This is a common discussion in our PRs.